tests/archive.rs

```diff
@@ -54,51 +53,47 @@ fn parse_archive() {
 }
 
 #[test]
-#[cfg(not(target_os="macos"))]  // pending https://github.com/m4b/goblin/issues/26
 fn parse_self() {
     use std::io::Read;
     let path = Path::new("target").join("debug").join("libgoblin.rlib");
-    match File::open(path) {
-        Ok(mut fd) => {
-            let buffer = { let mut v = Vec::new(); fd.read_to_end(&mut v).unwrap(); v};
-            match Archive::parse(&buffer) {
-                Ok(archive) => {
-                    let mut found = false;
-                    for member in archive.members() {
-                        println!("member: {:?}", member);
-                        if member.starts_with("goblin") &&
-                            // pre 1.18
-                            (member.ends_with("0.o") ||
-                             // >= 1.18
-                             member.ends_with("goblin-archive.o")) {
-                            match archive.extract(member.as_str(), &buffer) {
-                                Ok(bytes) => {
-                                    match elf::Elf::parse(&bytes) {
-                                        Ok(elf) => {
-                                            assert!(elf.entry == 0);
-                                            assert!(elf.bias == 0);
-                                            found = true;
-                                            break;
-                                        },
-                                        Err(err) => {
-                                            println!("{:?}", err);
-                                            #[cfg(target_os="linux")]
-                                            assert!(false)
-                                        }
-                                    }
-                                },
-                                Err(_) => assert!(false)
-                            }
-                        }
-                    }
-                    if !found {
-                        println!("goblin-<hash>.0.o not found");
-                        assert!(false)
-                    }
-                },
-                Err(err) => {println!("{:?}", err); assert!(false)}
-            }
-        },
-        Err(err) => {println!("{:?}", err); assert!(false)}
+    let buffer = {
+        let mut fd = File::open(path).expect("open file");
+        let mut v = Vec::new();
+        fd.read_to_end(&mut v).expect("read file");
+        v
+    };
+
+    let archive = Archive::parse(&buffer).expect("parse rlib");
+
+    // check that the archive has a useful symbol table
+    archive.summarize().into_iter()
+        .filter(|&(_member_name, _member_index, ref symbols)| symbols.len() > 500)
+        .next()
+        .expect("archive should contain a member with at least 500 symbols");
```

Comment:
Just FYI this is testing that there are at least 500 symbols in first member. In rustc 1.16 this will be probably be true as everything is a single object file. Later versions split up the rlib into multiple object files in the archive, so it will likely be untrue. Safest way to fix this for all versions is just sum the symbols.